File: apollo-runtime/src/main/java/com/apollographql/apollo/internal/cache/normalized/ReadableStore.java
package com.apollographql.apollo.internal.cache.normalized;

import com.apollographql.apollo.cache.CacheHeaders;
import com.apollographql.apollo.cache.normalized.Record;

import java.util.Collection;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public interface ReadableStore {

  @Nullable Record read(@NotNull String key, @NotNull CacheHeaders cacheHeaders);

  Collection<Record> read(@NotNull Collection<String> keys, @NotNull CacheHeaders cacheHeaders);

}

File: apollo-normalized-cache-api/src/commonMain/kotlin/com/apollographql/apollo/cache/normalized/internal/ReadableStore.kt
package com.apollographql.apollo.cache.normalized.internal

import com.apollographql.apollo.api.internal.json.JsonReader
import com.apollographql.apollo.cache.CacheHeaders
import com.apollographql.apollo.cache.normalized.Record

interface ReadableStore {
  fun read(key: String, cacheHeaders: CacheHeaders): Record?
  fun read(keys: Collection<String>, cacheHeaders: CacheHeaders): Collection<Record>
}
